File: liqp-core/src/main/java/liqp/filters/UrlDecode.java
package liqp.filters;

import java.net.URLDecoder;
import liqp.nodes.RenderContext;

public class UrlDecode extends Filter {

    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        try {
            return URLDecoder.decode(super.asString(value), "UTF-8");
        }
        catch (Exception e) {
            return value;
        }
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/UrlDecode.kt
package liqp.filter

import liqp.context.LContext
import liqp.params.FilterParams
import java.net.URLDecoder

class UrlDecode : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return try {
      URLDecoder.decode(context.asString(value), "UTF-8")
    } catch (e: Exception) {
      value
    }
  }
}
